Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeSwitcher.h

```diff
@@ -34,11 +34,11 @@ namespace AzToolsFramework
 
             // Member variables
             AZ::EntityComponentIdPair m_pairId; //!< Id of entity component pair.
-            AZ::Entity* m_entity = nullptr; //!< Pointer to entity associated with pairId.
-            AZ::Component* m_component = nullptr; //!< Pointer to component associated with pairId.
             AZStd::string m_componentName; //!< Friendly name of component.
             AZStd::string m_iconPath; //!< Path of component icon.
             ViewportUi::ButtonId m_buttonId; //!< Button Id of switcher component.
+
+            AZ::Component* FindComponent() const;
```

Comment:
nit: It's usually good to return a `const` pointer from a const member function (or have an overload so you have...)

```c++
AZ::Component* FindComponent();
const AZ::Component* FindComponent() const;
```

If you just need the `const` one then don't bother with the non-const overload.

This is referred to as 'const broken', this page has a nice overview - https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/const.md#classes-of-const-in_correctness